js/components/common/__tests__/CourseModalButtons.test.js

```diff
@@ -0,0 +1,75 @@
+import { screen, waitFor, fireEvent } from "@testing-library/react"
+import TestGraph from "../../graph/__tests__/TestGraph.js"
+
+describe("CourseModal", () => {
+  beforeEach(async () => {
+    await TestGraph.build()
+
+    // find the course node AAA100 and open its info modal
+    const node = document.querySelector('[data-testid="aaa100"]')
+    fireEvent.mouseOver(node)
+    const infobox = document.getElementById("aaa100-tooltip-rect")
+    fireEvent.click(infobox)
+  })
+
+  // wait for the modal to appear
+  const waitForModalUpdate = async () => {
+    await waitFor(() => {
+      const updatedModal = document.querySelector(".ReactModalPortal")
+      return updatedModal
+    })
+  }
+
+  it("does not render buttons when course modal is initially opened", async () => {
+    await waitForModalUpdate()
+
+    // the back and forward buttons do not appear
+    const buttons = document.querySelectorAll(".info-modal-button")
+    expect(buttons.length === 0).toBe(true)
+  })
+
+  it("renders buttons when a course link is clicked", async () => {
+    await waitForModalUpdate()
+
+    // click on the course link BBB100H1
+    const courseLink = screen.getByText("BBB100H1")
+    fireEvent.click(courseLink)
+
+    // the back and forward buttons render
+    const buttons = document.querySelectorAll(".info-modal-button")
+    expect(buttons.length == 2).toBe(true)
+
+    // the forward button is disabled, but the back button is enabled
+    const forwardButton = screen.getByText(">")
+    const backButton = screen.getByText("<")
+    expect(forwardButton.disabled).toBe(true)
```

Comment:
You are getting at some good test cases here, but should create additional test cases for all four possible combinations of enabled/disabled states for the two buttons